modules/programs/zsh/default.nix

```diff
@@ -343,6 +345,30 @@ in
     in
     mkIf cfg.enable (
       lib.mkMerge [
+        {
+          assertions = [
+            {
+              assertion = !lib.hasInfix "$" cfg.dotDir;
```

Comment:
Technically you can have a directory with `$` in the name... not sure why you would though.

Reply:
Don't want to deal with the assumption, just going to remove.

Reply:
Hmm.. maybe you're right. I could add it back if you think it provides more value than risk...

Reply:
I meant I like having this assertion (which you didn't remove), because it prevents users from using `$` in `dotDir`, where most likely they wanted to use an env var instead of a `$` in the folder name. I think the test case you removed in https://github.com/nix-community/home-manager/compare/9ba29a1bc7f64b3803420372928c0d6f5989522c..6fba8803792b709e3a3c166f5b7aea480515d0b7 is not needed anymore, as `dotDir` can't contain a `$`

Reply:
> I meant I like having this assertion (which you didn't remove), because it prevents users from using `$` in `dotDir`, where most likely they wanted to use an env var instead of a `$` in the folder name. I think the test case you removed in https://github.com/nix-community/home-manager/compare/9ba29a1bc7f64b3803420372928c0d6f5989522c..6fba8803792b709e3a3c166f5b7aea480515d0b7 is not needed anymore, as `dotDir` can't contain a `$`

The assertion was removed. That's just the latest force push cleaning up tests after yanking out the assertion.